anndata/core/anndata.py

```diff
@@ -489,38 +487,10 @@ def _init_as_actual(
         self._check_dimensions()
         self._check_uniqueness()
 
-        # raw
-        if raw is None:
-            self._raw = None
-        elif isinstance(raw, Raw):
-            self._raw = raw
-        elif isinstance(raw, cabc.Mapping):
```

Comment:
Removing the parameter should be fine:

1. `AnnData(raw=None)` is useless for users
2. `AnnData(raw=dict(...))` is an internal API and has to go
3. `AnnData(raw=adata2.raw)` might be OK, we need to evaluate if people use this.

Bonus: Allowing `AnnData(raw=adata2)` would make most sense, as `raw.setter` already only accepts AnnDatas.

But generally, you only want `.raw` if you want to save an AnnData object without losing yout raw data, right? So just having `adata.raw = ...` should be enough.

Reply:
I think having raw be a dict would be okay as long as it's practically `self.raw = Raw(**raw)`. I think mutable mappings are a pretty useful interface, and I like having this nice recursive definition available to the user.